engine/db/db_server.cpp

```diff
@@ -133,6 +134,57 @@ Status DBServer::Insert(const std::string& db_name,
   return table->Insert(records);
 }
 
+Status DBServer::DeleteByPK(const std::string& db_name, const std::string& table_name, vectordb::Json& pkList) {
+  auto db = GetDB(db_name);
+  if (db == nullptr) {
+    return Status(DB_UNEXPECTED_ERROR, "DB not found: " + db_name);
+  }
+  auto table = db->GetTable(table_name);
+  if (table == nullptr) {
+    return Status(DB_UNEXPECTED_ERROR, "Table not found: " + table_name);
+  }
+  int pkIdx = 0;
+  for (; pkIdx < table->table_schema_.fields_.size(); pkIdx++) {
+    if (table->table_schema_.fields_[pkIdx].is_primary_key_) {
+      break;
+    }
+  }
+  if (pkIdx == table->table_schema_.fields_.size()) {
+    return Status(DB_UNEXPECTED_ERROR, "PK not found: " + table_name);
```

Comment:
"Primary key not found"

Reply:
92dc14967714effffe4611e3103d35b9002d6a93